Can't build KVirc from git on Arch Linux

The `kvirc-git` AUR package fails to build with the following error:

```
In file included from /usr/include/python3.5m/pytime.h:6:0,
                 from /usr/include/python3.5m/Python.h:65,
                 from /home/vladimir/work/extern/arch/aur/kvirc-git/src/kvirc-git/src/modules/pythoncore/libkvipythoncore.cpp:37:
/usr/include/python3.5m/object.h:445:23: error: expected unqualified-id before ‘;’ token
     PyType_Slot *slots; /* terminated by slot==0. */
                       ^
/home/vladimir/work/extern/arch/aur/kvirc-git/src/kvirc-git/src/modules/pythoncore/kvircmodule.cpp: In function ‘PyObject* python_init()’:
/home/vladimir/work/extern/arch/aur/kvirc-git/src/kvirc-git/src/modules/pythoncore/kvircmodule.cpp:426:72: error: ‘PyCObject_FromVoidPtr’ was not declared in this scope
   PyObject * pC_API_Object = PyCObject_FromVoidPtr(PyKVIrc_API, nullptr);
                                                                        ^
```

```
~/work/extern/arch/aur/kvirc-git/my/KVIrc-build » cmake ../KVIrc 
-- The C compiler identification is GNU 6.1.1
-- The CXX compiler identification is GNU 6.1.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.9.0") 
-- Found Git 2.9.0
-- Current revision is git-6748-g6804848
-- Using sources date from 20160621
-- Performing Test CXX_HAS_VIH_FLAG
-- Performing Test CXX_HAS_VIH_FLAG - Success
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of short
-- Check size of short - done
-- Check size of int
-- Check size of int - done
-- Check size of long
-- Check size of long - done
-- Check size of long long
-- Check size of long long - done
-- Looking for include file sys/types.h
-- Looking for include file sys/types.h - found
-- Looking for include file strings.h
-- Looking for include file strings.h - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found ZLIB: /usr/lib64/libz.so (found version "1.2.8") 
-- Found KF5CoreAddons: /usr/lib64/cmake/KF5CoreAddons/KF5CoreAddonsConfig.cmake (found version "5.23.0") 
-- Found Gettext: /usr/bin/msgmerge (found version "0.19.7") 
-- Found PythonInterp: /usr/bin/python (found version "3.5.1") 
-- Found KF5I18n: /usr/lib64/cmake/KF5I18n/KF5I18nConfig.cmake (found version "5.23.0") 
-- Found KF5XmlGui: /usr/lib64/cmake/KF5XmlGui/KF5XmlGuiConfig.cmake (found version "5.23.0") 
-- Found KF5WindowSystem: /usr/lib64/cmake/KF5WindowSystem/KF5WindowSystemConfig.cmake (found version "5.23.0") 
-- Found KF5Notifications: /usr/lib64/cmake/KF5Notifications/KF5NotificationsConfig.cmake (found version "5.23.0") 
-- Found KF5Service: /usr/lib64/cmake/KF5Service/KF5ServiceConfig.cmake (found version "5.23.0") 
-- Found KF5: success (found suitable version "5.23.0", minimum required is "5.2.0") found components:  CoreAddons I18n XmlGui WindowSystem Notifications Service 
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so
-- Looking for XOpenDisplay in /usr/lib64/libX11.so;/usr/lib64/libXext.so - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for connect
-- Looking for connect - found
-- Looking for remove
-- Looking for remove - found
-- Looking for shmat
-- Looking for shmat - found
-- Looking for IceConnectionNumber in ICE
-- Looking for IceConnectionNumber in ICE - found
-- Found X11: /usr/lib64/libX11.so
-- Found OpenSSL: /usr/lib64/libssl.so;/usr/lib64/libcrypto.so (found version "1.0.2h") 
-- Found Perl: /usr/bin/perl (found version "5.24.0") 
-- Found PythonLibs: /usr/lib64/libpython3.5m.so (found suitable version "3.5.1", minimum required is "2.7") 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1") 
-- Checking for module 'enchant'
--   Found enchant, version 1.6.0
-- Looking for inet_ntoa
-- Looking for inet_ntoa - found
-- Looking for inet_aton
-- Looking for inet_aton - found
-- Looking for setenv
-- Looking for setenv - found
-- Looking for putenv
-- Looking for putenv - found
-- Looking for linux/soundcard.h
-- Looking for linux/soundcard.h - found
-- Looking for sys/soundcard.h
-- Looking for sys/soundcard.h - found
-- Looking for soundcard.h
-- Looking for soundcard.h - not found
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.11") 
--  
-- OK, this is the configuration I've worked out:
--  
-- Paths:
--    Binary path                 : /usr/local/bin/
--    Library path                : /usr/local/lib/
--    Modules path                : /usr/local/lib/kvirc/4.9/modules/
--    Man path                    : /usr/local/share/man/
-- Compilation:
--    Build type                  : Release
--    Debug                       : No (Release)
--    Verbose output              : No
--    Coexistence support         : No
--    Threading support           : POSIX
--    Apple universal binary      : Not available
--    Memory profile support      : No
--    Memory checks support       : No
-- Features:
--    X11 support                 : Yes, with scrnsaveproto
--    Qt version                  : 5.6.1
--    Qt modules                  : Widgets;Sql;Xml;Network;PrintSupport;Multimedia;DBus;WebKitWidgets;Svg;X11Extras
--    Qt/X11 interface            : Yes
--    KDE support                 : Yes
--    KDE version                 : 5.23.0
--    IPv6 support                : Yes
--    Pseudo-transparency support : Yes
--    Spell-checker support       : Yes
--    IPC support                 : Yes
--    Audio backends              : Phonon;Qt5Multimedia;Oss+AudioFile
-- Cryptography:
--    Crypt support and engines   : Yes
--    OpenSSL support             : Yes
-- Scripting:
--    Perl support                : Yes, used for scripting+doc
--    Python support              : Yes
-- DCC Features:
--    DCC voice support           : Yes, found linux/soundcard.h
--    GSM support                 : Yes
--    DCC video support           : No
--    Ogg/Theora support          : No
-- Documentation:
--    gettext support             : Yes, + extra
--    gettext messages tidying    : Yes
--    Doxygen support             : Yes
--  
-- Build date                     : 2016-06-24 13:24:07 UTC
-- Build version                  : 4.9.2
-- Build codename                 : Aria
-- Build revision                 : git-6748-g6804848
-- Sources date                   : 20160621
--  
-- I'm ready to build KVIrc for you: please type 'make' now...
--  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/vladimir/work/extern/arch/aur/kvirc-git/my/KVIrc-build

```

The issue is reproducible when built by hand following the `INSTALL.txt` instructions.

An excerpt from the CMake output:

```
-- Found PythonLibs: /usr/lib64/libpython3.5m.so (found suitable version "3.5.1", minimum required is "2.7")
```

This appears to be a regression. Bisecting shows that the regression was introduced in https://github.com/kvirc/KVIrc/commit/a589f834e10f8767cbfb7ac26485cfcd07ff2214

With that commit reverted, KVirc builds successfully. The relevant CMake line now states:

```
-- Found PythonLibs: /usr/lib64/libpython2.7.so (found version "2.7.11") 
```

It's not clear whether the intention of that commit was to restrict the Python version to exclude only Python versions before 2.7, or also 3.x as well. [CMake's documentation on `find_package`](https://cmake.org/cmake/help/v3.0/command/find_package.html) is vague:

> The [version] argument requests a version with which the package found should be compatible

CC @un1versal @staticfox @Dessa

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,6 +69,9 @@ set(CMAKE_KVIRC_BUILD_CPU ${CMAKE_SYSTEM_PROCESSOR})
 set(CMAKE_KVIRC_BUILD_COMPILER ${CMAKE_CXX_COMPILER})
 set(CMAKE_KVIRC_BUILD_COMPILER_FLAGS ${CMAKE_CXX_FLAGS})
 
+# Prefer Python 2.7 over 3.x (which is currently incompatible) - GitHub issue #2020
+set(Python_ADDITIONAL_VERSIONS "2.7")
+
 # Allow the use of the LOCATION property
 cmake_policy(SET CMP0026 OLD)
 

